File: src/net/merayen/elastic/backend/logicnodes/list/group_1/LogicNode.java
package net.merayen.elastic.backend.logicnodes.list.group_1;

import java.util.Map;

import net.merayen.elastic.backend.nodes.BaseLogicNode;

/**
 * Doesn't do anything, other than having children.
 * @author merayen
 *
 */
public class LogicNode extends BaseLogicNode {
	@Override
	protected void onCreate() {}

	@Override
	protected void onInit() {}

	@Override
	protected void onParameterChange(String key, Object value) {}

	@Override
	protected void onConnect(String port) {}

	@Override
	protected void onDisconnect(String port) {}

	@Override
	protected void onRemove() {}

	@Override
	protected void onPrepareFrame(Map<String, Object> data) {}

	@Override
	protected void onFinishFrame(Map<String, Object> data) {}

	@Override
	protected void onData(Map<String, Object> data) {}

}

File: src/main/kotlin/net/merayen/elastic/backend/logicnodes/list/data_1/LogicNode.kt
package net.merayen.elastic.backend.logicnodes.list.data_1

import net.merayen.elastic.backend.nodes.BaseLogicNode
import net.merayen.elastic.system.intercom.OutputFrameData

class LogicNode : BaseLogicNode() {
	override fun onCreate() {}
	override fun onInit() {}
	override fun onData(data: Any) {}
	override fun onConnect(port: String?) {}
	override fun onDisconnect(port: String?) {}
	override fun onRemove() {}
	override fun onPrepareFrame(data: MutableMap<String, Any>?) {}
	override fun onFinishFrame(data: OutputFrameData) {}

	override fun onParameterChange(key: String?, value: Any?) {
		set(key, value)
	}
}